### ISSUE TITLE
More then one matching SLURM task

### ISSUE THREAD

**Issue by albertz** [rwth-i6/sisyphus#156]
```
[2023-11-23 12:24:05,584] ERROR: Error to submit job                                                                                           
[2023-11-23 12:24:05,584] ERROR: SBATCH command: sbatch -J i6_core.returnn.forward.ReturnnForwardJobV2.9mzWqrCWqG0E.run -o work/i6_core/returnn
/forward/ReturnnForwardJobV2.9mzWqrCWqG0E/engine/%x.%A.%a --mail-type=None --mem=6G --gres=gpu:1 --cpus-per-task=2 --time=240 --export=all -p g
pu_11gb -x cn-505 -a 1-1:1 --wrap=/work/tools/users/zeyer/py-envs/py3.11-torch2.1/bin/python3.11 ./sis worker --engine long work/i6_core/return
n/forward/ReturnnForwardJobV2.9mzWqrCWqG0E run                                                                                                 
[2023-11-23 12:24:05,585] ERROR: Output: Submitted batch job 3149172                                                                           
[2023-11-23 12:24:05,585] ERROR: Error: sbatch: error: Slurm temporarily unable to accept job, sleeping and retrying                           
0 1 4 [b'Submitted', b'batch', b'job'] [b'Submitted', b'batch', b'job']                                                                        
[2023-11-23 12:24:05,586] ERROR: Error to submit job                                                                                           
[2023-11-23 12:24:05,586] ERROR: SBATCH command: sbatch -J i6_core.returnn.forward.ReturnnForwardJobV2.xsJCcIe5eUm7.run -o work/i6_core/returnn
/forward/ReturnnForwardJobV2.xsJCcIe5eUm7/engine/%x.%A.%a --mail-type=None --mem=6G --gres=gpu:1 --cpus-per-task=2 --time=240 --export=all -p g
pu_11gb -x cn-505 -a 1-1:1 --wrap=/work/tools/users/zeyer/py-envs/py3.11-torch2.1/bin/python3.11 ./sis worker --engine long work/i6_core/return
n/forward/ReturnnForwardJobV2.xsJCcIe5eUm7 run                                                                                                 
[2023-11-23 12:24:05,586] ERROR: Output: Submitted batch job 3149173                                                                           
[2023-11-23 12:24:05,586] ERROR: Error: sbatch: error: Slurm temporarily unable to accept job, sleeping and retrying                           
0 1 4 [b'Submitted', b'batch', b'job'] [b'Submitted', b'batch', b'job']                                                                        
[2023-11-23 12:24:05,587] ERROR: Error to submit job                                                                                           
[2023-11-23 12:24:05,587] ERROR: SBATCH command: sbatch -J i6_core.returnn.forward.ReturnnForwardJobV2.vinrAG5lsR8a.run -o work/i6_core/returnn
/forward/ReturnnForwardJobV2.vinrAG5lsR8a/engine/%x.%A.%a --mail-type=None --mem=6G --gres=gpu:1 --cpus-per-task=2 --time=240 --export=all -p g
pu_11gb -x cn-505 -a 1-1:1 --wrap=/work/tools/users/zeyer/py-envs/py3.11-torch2.1/bin/python3.11 ./sis worker --engine long work/i6_core/return
n/forward/ReturnnForwardJobV2.vinrAG5lsR8a run                                                                                                 
[2023-11-23 12:24:05,587] ERROR: Output: Submitted batch job 3149174                                                                           
[2023-11-23 12:24:05,587] ERROR: Error: sbatch: error: Slurm temporarily unable to accept job, sleeping and retrying                           
...
[2023-11-23 12:24:48,575] WARNING: More then one matching SLURM task, use first match < ('i6_core.returnn.forward.ReturnnForwardJobV2.q6p2w4D9C
SYW.run', 1) > matches: [('3149158', 'PENDING'), ('3149194', 'PENDING')]                                 
[2023-11-23 12:24:48,580] WARNING: More then one matching SLURM task, use first match < ('i6_core.returnn.forward.ReturnnForwardJobV2.IAkRxmOdW
HHA.run', 1) > matches: [('3149156', 'PENDING'), ('3149197', 'PENDING')]                                 
[2023-11-23 12:24:48,597] WARNING: More then one matching SLURM task, use first match < ('i6_core.returnn.forward.ReturnnForwardJobV2.lkURo50Hf
4d2.run', 1) > matches: [('3149155', 'PENDING'), ('3149198', 'PENDING')]                                 
[2023-11-23 12:24:48,636] WARNING: More then one matching SLURM task, use first match < ('i6_core.returnn.forward.ReturnnForwardJobV2.iaKfugRIW
bb7.run', 1) > matches: [('3149157', 'PENDING'), ('3149200', 'PENDING')]                                 
[2023-11-23 12:24:48,665] WARNING: More then one matching SLURM task, use first match < ('i6_core.returnn.forward.ReturnnForwardJobV2.jDBAArsaJ
aWc.run', 1) > matches: [('3149159', 'PENDING'), ('3149199', 'PENDING')]                                 
[2023-11-23 12:24:48,675] WARNING: More then one matching SLURM task, use first match < ('i6_core.returnn.forward.ReturnnForwardJobV2.QaE4jVxus
6xu.run', 1) > matches: [('3149161', 'PENDING'), ('3149196', 'PENDING')]                                 
[2023-11-23 12:24:48,708] WARNING: More then one matching SLURM task, use first match < ('i6_core.returnn.forward.ReturnnForwardJobV2.YOMu5k2SC
FZi.run', 1) > matches: [('3149160', 'PENDING'), ('3149195', 'PENDING')]                                 
```

I assume that Sisyphus somehow did not get the full output status, and the incorrectly just assumed that the jobs don't exist, and resubmitted them.

The fix would be to never submit jobs when the current job status is unknown.

@Atticus1806 also reported this.

**Comment by Atticus1806** [comment]
I never investigated the reason yet, but anecdotally this is the type of error message that occurs beforehand, not sure if related:

```
[2023-11-23 10:52:11,445] ERROR: Error to submit job                                                                                                                                                                                                                               [2023-11-23 10:52:11,445] ERROR: SBATCH command: sbatch -J LatticeToCtmJob.J6rwzyknMo98.run -o work/i6_core/recognition/conversion/LatticeToCtmJob.J6rwzyknMo98/engine/%x.%A.%a --mail-type=None --mem=4G --cpus-per-task=1 --time=30 --export=all -a 1-5:1 --wrap=apptainer exec -B /work/asr4 -B /work/asr3 -B /work/common -B /work/tools -B /work/speech /work/asr4/hilmes/apptainer/u22_torch_onnx_115.sif sis worker --engine long work/i6_core/recognition/conversion/LatticeToCtmJob.J6rwzyknMo98 run                                                         
[2023-11-23 10:52:11,445] ERROR: Output: Submitted batch job 3146028                                                                                                                                                                                                               [2023-11-23 10:52:11,445] ERROR: Error: sbatch: error: Slurm temporarily unable to accept job, sleeping and retrying
```

**Comment by michelwi** [comment]
This log is coming from https://github.com/rwth-i6/sisyphus/blob/450ed58ec6ec964db0b8377b2b685d37b77fa3db/sisyphus/simple_linux_utility_for_resource_management_engine.py#L237-L255

i.e.  `Error: sbatch: error: Slurm temporarily unable to accept job, sleeping and retrying` is coming from the slurm engine and sisyphus interpretes this as a failure to submit, while, I guess, slurm (silently?) resubmits the job later.

We could either
* add a check in this part of the code and if the "error" that slurm throws contains "sleeping and retrying" count it as a success.
* configure Slurm somehow to abort the submission if not immediately successfull
* check, why slurm is hanging and fix the underlying issue.

**Comment by dthulke** [comment]
```
Error: sbatch: error: Slurm temporarily unable to accept job, sleeping and retrying
```

I just looked into the together with @NeoLegends as he had problems with a setup in which he submitted a lot of jobs.
The solution seemed to be to enable the `defer_batch` option in slurm so that jobs are not scheduled directly but can be deferred (see https://slurm.schedmd.com/slurm.conf.html#OPT_defer).
This fixed it in his setup and so maybe the underlying problem is solved.

**Comment by michelwi** [comment]
> enable the defer_batch option in slurm

is this something that sisyphus should do (i.e. an option to sbatch) or some configuration on cluster level?

Do you know what is the maximum delay that a submission would be deferred? We might have to adapt one of
https://github.com/rwth-i6/sisyphus/blob/450ed58ec6ec964db0b8377b2b685d37b77fa3db/sisyphus/global_settings.py#L184-L187
accordingly.

**Comment by albertz** [comment]
Note, from the output:
```
0 1 4 [b'Submitted', b'batch', b'job'] [b'Submitted', b'batch', b'job']                                                                        
[2023-11-23 12:24:05,586] ERROR: Error to submit job                                                                                           
[2023-11-23 12:24:05,586] ERROR: SBATCH command: sbatch -J i6_core.returnn....
[2023-11-23 12:24:05,586] ERROR: Output: Submitted batch job 3149173                                                                           
[2023-11-23 12:24:05,586] ERROR: Error: sbatch: error: Slurm temporarily unable to accept job, sleeping and retrying                           
```
You see:
It actually successfully submitted the job? There is the output "Submitted batch job 3149173". Also we got the exit code 0, for success. So maybe this error can be ignored in this case?

**Comment by NeoLegends** [comment]
The output is interleaved from many threads, so the job ID you see there does not correspond to the error messages in the log.

**Comment by albertz** [comment]
> The output is interleaved from many threads (sisyphus uses a thread pool to submit jobs), so the job ID you see there does not correspond to the error messages in the log.

No, you see also from the `print` before that it indeed has this output on stdout. So I'm pretty sure it successfully submitted the job.

**Comment by albertz** [comment]
I have a PR to fix this: #157
